src/api/define.js

```diff
@@ -102,10 +102,19 @@ function generateUniqueName(name) {
 }
 
 function registerUniqueName(name) {
-  registry[name] = registry[name] ? registry[name] + 1 : 1;
+  if (registry[name]) {
+    registerUniqueName(name + '-'+ registry[name]);
```

Comment:
I can see how someone could get caught up by that and am now seeing extra step vs recursion because you're incrementing below. 👍